### PR TITLE
166240826 create article endpoint 

### PR DESCRIPTION
**Pull request by murediane**
#### use case document
https://docs.google.com/document/d/19dedbuAmUxZC5wIhQvPtRvOEYkuN_HRsZDKj89Twgp4/edit
#### What does this PR do?
create article endpoint 
#### Description of Task to be completed?
create article endpoint 
#### How should this be manually tested?
run the server and send a request using postman on api/articles
#### Any background context you want to provide?
Adapt authorization to be used with all routes to avoid having different middleware doing the same thing
#### What are the relevant pivotal tracker stories?
[166240826](https://www.pivotaltracker.com/story/show/166240826)
#### Screenshots (if appropriate)
createres
createreq,res

### CODE REVIEW

**Comment by nkpremices** [comment]
> #### use case document
> https://docs.google.com/document/d/19dedbuAmUxZC5wIhQvPtRvOEYkuN_HRsZDKj89Twgp4/edit
> 
> #### What does this PR do?
> create article endpoint
> 
> #### Description of Task to be completed?
> checkauth middleware to be able to secure the create article route
> create article endpoint
> 
> #### How should this be manually tested?
> run the server and send a request using postman on api/articles
> 
> #### Any background context you want to provide?
> #### What are the relevant pivotal tracker stories?
> [166240826](https://www.pivotaltracker.com/story/show/166240826)
> 
> #### Screenshots (if appropriate)
> createres
> 
> createreq,res

* Try to put bullets @murediane 
* what does `checkauth middleware to be able to secure the create article route` mean ?

File: src/api/controllers/articleController.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const { Article, Category } = models;
+const { CREATED } = statusCodes;
+/**
+ * containing all article's model controllers
+ * @export
+ * @class ArticleController
+ */
+export default class ArticleController {
+  /**
+   * register a new
```
Comment (premlogic) [inline src/api/controllers/articleController.js:25]: `A controller to create a new article`

File: src/api/controllers/articleController.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+   * @memberof ArticleController
+   * @returns {Object} res
+   */
+  static async create(req, res) {
+    const { id: author } = req.user;
+    const valid = await articleValidator(req.body);
+    const { category: name } = req.body;
+    let image = '';
+
+    if (req.file) {
```
Comment (premlogic) [inline src/api/controllers/articleController.js:38]: If the process is passing through a middleware, you don't need this if close here. 
My point is, it will never reach the controller if it violates something in the middleware @murediane

File: src/api/controllers/articleController.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      const saved = await cloudinary.v2.uploader.upload(req.file.path);
```
Comment (premlogic) [inline src/api/controllers/articleController.js]: Try to use `easily understandable variable names`

File: src/api/controllers/articleController.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      image = saved.secure_url;
+    }
+    await Category.findOrCreate({ where: { name } }).spread(async (cat) => {
+      const { id: category } = cat.get();
+      const article = await Article.create({
+        ...valid,
```
Comment (premlogic) [inline src/api/controllers/articleController.js]: It's not easy to understand what's happening here just by looking at the `valid` variable name

File: src/api/controllers/authController.js [hunk 5]
```diff
@@ -39,9 +39,9 @@ export default class Auth {
       email,
```
Comment (premlogic) [inline src/api/controllers/authController.js:39]: please @murediane  remove this file from your PR. It's not related to it

File: src/api/migrations/20190620193548-create-token.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+export default {
```
Comment (premlogic) [inline src/api/migrations/20190620193548-create-token.js]: this migration is no longer used. 
I think @MCFrank16 updated it. He created a migration to delete this table

File: src/api/migrations/20190622195901-create-category.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+export default {
+  up: (queryInterface, Sequelize) => queryInterface.createTable('categories', {
```
Comment (premlogic) [inline src/api/migrations/20190622195901-create-category.js]: The table name should start with a capital letter @murediane

File: src/api/models/article.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
+import slug from 'slug';
+
+const THIRTY_SIX = 36;
```
Comment (premlogic) [inline src/api/models/article.js]: @MemunaHaruna gave feedback on this @murediane .
Use variable names that reflect what you used them for

File: src/api/models/category.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        primaryKey: true,
+        type: DataTypes.INTEGER
+      },
+      name: { type: DataTypes.STRING, allowNull: false }
+    },
+    {
+      tableName: 'categories',
+      hooks: {
+        beforeCreate(category) {
+          category.name = category.name.toUpperCase();
```
Comment (premlogic) [inline src/api/models/category.js:16]: Why do we want to uppercase the category @murediane ?

File: src/api/models/token.js [hunk 10]
```diff
@@ -0,0 +1,8 @@
+export default (sequelize, DataTypes) => {
```
Comment (premlogic) [inline src/api/models/token.js]: this file shouldn't be on this PR @murediane

File: src/api/routes/articles.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+import checkAuth from '../../middlewares/checkAuth';
+import ArticleController from '../controllers/articleController';
+import errorHandler from '../../middlewares/errorHandler';
+import uploadImage from '../../middlewares/upload';
+
+const articleRouter = new Router();
+
+articleRouter
+  .route('/articles')
+  .post(checkAuth, uploadImage.single('image'), errorHandler(ArticleController.create));
```
Comment (premlogic) [inline src/api/routes/articles.js]: check also if you are receiving an empty object.

Comment (premlogic) [inline src/api/routes/articles.js]: I didn't see any validation for this endpoint @murediane

File: src/api/routes/index.js [hunk 12]
```diff
@@ -1,10 +1,12 @@
 import express from 'express';
 import authRouter from './authRouter';
 import resetRouter from './password.reset';
+import articleRouter from './articles';
 
 const router = express();
 
 router.use('/users', authRouter);
 router.use('/password-reset', resetRouter);
+router.use(articleRouter);
```
Comment (premlogic) [inline src/api/routes/index.js]: I think it makes more sense if you use `router.use('/articles', articleRouter);` @murediane

File: src/api/routes/articles.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import { Router } from 'express';
+import checkAuth from '../../middlewares/checkAuth';
+import ArticleController from '../controllers/articleController';
+import errorHandler from '../../middlewares/errorHandler';
+import uploadImage from '../../middlewares/upload';
+
+const articleRouter = new Router();
+
+articleRouter
+  .route('/articles')
```
Comment (premlogic) [inline src/api/routes/articles.js]: as we'll all use this router, it's better to use this route in the `index.js` then use `.post(checkAuth, uploadImage.single('image')`

File: package.json [hunk 14]
```diff
@@ -5,12 +5,14 @@
   "main": "index.js",
   "scripts": {
     "start": "babel-node src/index.js",
-    "test": "npm run setup:test && NODE_ENV=test nyc --reporter=html --reporter=text --reporter=lcov ./node_modules/.bin/mocha  ./tests/* --require @babel/register --timeout 4000 --exit",
+    "test": "NODE_ENV=test nyc --reporter=html --reporter=text --reporter=lcov ./node_modules/.bin/mocha  ./tests/* --require @babel/register --timeout 4000 --exit",
     "dev": "NODE_ENV=development nodemon --exec babel-node src/index.js",
```
Comment (alainburindi) [inline package.json:9]: what did you change the test script??

File: src/middlewares/checkAuth.js [hunk 15]
```diff
@@ -0,0 +1,11 @@
+import passport from 'passport';
+import statusCodes from '../helpers/constants/status.codes';
+import errorMessages from '../helpers/constants/error.messages';
+
+const { UNAUTHORIZED } = statusCodes;
```
Comment (premlogic) [inline src/middlewares/checkAuth.js]: does this use `Bearer` ?

File: src/middlewares/errorHandler.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
+import statusCodes from '../helpers/constants/status.codes';
```
Comment (premlogic) [inline src/middlewares/errorHandler.js:1]: Great Idea @murediane

File: src/api/migrations/20190622195901-create-category.js [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+    id: {
```
Comment (alainburindi) [inline src/api/migrations/20190622195901-create-category.js:3]: can you please use a capital letter at the beginning of the table name? i.e. : Categories

File: src/middlewares/errorHandler.js [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+const { BAD_REQUEST, SERVER_ERROR } = statusCodes;
+const { serverError } = errorMessages;
+
+export default callback => async (req, res, next) => {
+  try {
+    await callback(req, res, next);
+  } catch (error) {
+    console.log(error);
+    const { parent, details } = error;
+    if (parent) return res.status(SERVER_ERROR).json({ message: parent.detail });
```
Comment (premlogic) [inline src/middlewares/errorHandler.js]: Use the error sender @murediane

File: src/middlewares/checkAuth.js [hunk 19]
```diff
@@ -0,0 +1,11 @@
...
+const { authenticationMessage } = errorMessages;
+export default (req, res, next) => passport.authenticate('jwt', { session: false }, (err, user) => {
+  if (!user) return res.status(UNAUTHORIZED).json({ message: authenticationMessage });
```
Comment (premlogic) [inline src/middlewares/checkAuth.js]: Use the `error sender` @murediane

File: src/api/models/article.js [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+const SIX = 6;
+const ZERO = 0;
+
```
Comment (alainburindi) [inline src/api/models/article.js:6]: use described variable name

File: src/middlewares/passport.js [hunk 21]
```diff
@@ -0,0 +1,19 @@
+import { Strategy, ExtractJwt } from 'passport-jwt';
```
Comment (premlogic) [inline src/middlewares/passport.js]: I like the Idea of using passport to validate our tokens. But as we are not . using the tokens from social media, why do you use passport just to validate a token?

File: src/middlewares/upload.js [hunk 22]
```diff
@@ -0,0 +1,14 @@
+import multer from 'multer';
```
Comment (premlogic) [inline src/middlewares/upload.js]: what does this file do ?

File: src/api/models/index.js [hunk 23]
```diff
@@ -8,6 +8,9 @@ const sequelize = new Sequelize(env.dbUrl, {
 
 const models = {
   User: sequelize.import('./user'),
+  Token: sequelize.import('./token'),
```
Comment (alainburindi) [inline src/api/models/index.js]: we no longer have the token model

**Review by nkpremices — CHANGES_REQUESTED** [review]

**Comment by nkpremices** [comment]
@murediane 
I left some comments for you

File: src/helpers/cloudnary.js [hunk 24]
```diff
@@ -0,0 +1,10 @@
+import dotenv from 'dotenv';
+import cloudinary from 'cloudinary';
+
+dotenv.config();
+
+cloudinary.config({
```
Comment (alainburindi) [inline src/helpers/cloudnary.js]: can you please put this as the CLOUDINARY_URL in .env?? so that we may have the same structure

File: src/api/controllers/articleController.js (hunk 2 above)
Comment (murediane) [reply src/api/controllers/articleController.js:38]: I am not verifying if the image exists. the middleware is to help upload the file .here I am checking if the file exists.

File: src/api/controllers/articleController.js (hunk 4 above)
Comment (murediane) [reply src/api/controllers/articleController.js]: do you think validInput is better?

File: src/index.js [hunk 25]
```diff
@@ -1,18 +1,35 @@
...
+app.use(logger('dev'));
+
+app.use(express.static(`${__dirname}/public`));
+app.use(passport.initialize());
+passportConfig(passport);
 
-// Register middleware
-register(app);
+app.use(session({
+  secret: env.secret,
```
Comment (alainburindi) [inline src/index.js]: do we need session? if yes, why?

File: src/middlewares/passport.js (hunk 21 above)
Comment (murediane) [reply src/middlewares/passport.js]: I am not validating token, I'm decoding the token both while using social media or signup on the app by creating account

File: src/middlewares/upload.js (hunk 22 above)
Comment (alainburindi) [reply src/middlewares/upload.js]: it allows using form/multipath while sending the request so that we can get files

Comment (murediane) [reply src/middlewares/upload.js]: uploading file

File: src/api/controllers/authController.js (hunk 5 above)
Comment (murediane) [reply src/api/controllers/authController.js:39]: I terminated rebase after forgetting to save it .that is why `I had to add it but I did not really touch it .It is a mistake I did of forgetting saving the changes while in rebase mode

File: src/middlewares/errorHandler.js (hunk 16 above)
Comment (murediane) [reply src/middlewares/errorHandler.js:1]: thanks

**Review by alainburindi — CHANGES_REQUESTED** [review]

File: src/middlewares/errorHandler.js (hunk 18 above)
Comment (murediane) [reply src/middlewares/errorHandler.js]: thanks, working on it right now

File: src/middlewares/checkAuth.js (hunk 19 above)
Comment (murediane) [reply src/middlewares/checkAuth.js]: alright ,I am going to  work on it

File: src/middlewares/upload.js (hunk 22 above)
Comment (murediane) [reply src/middlewares/upload.js]: Thanks @alainburindi, I think that is clearer than my explanation

File: src/helpers/cloudnary.js (hunk 24 above)
Comment (murediane) [reply src/helpers/cloudnary.js]: It was conflicting with your Url .and i could not continue uploading using your account .I was getting access denied error

File: src/api/controllers/articleController.js [hunk 26]
```diff
@@ -0,0 +1,50 @@
...
+    const valid = await articleValidator(req.body);
+    const { category: name } = req.body;
+    let image = '';
+
+    if (req.file) {
+      const saved = await cloudinary.v2.uploader.upload(req.file.path);
+      image = saved.secure_url;
+    }
+    await Category.findOrCreate({ where: { name } }).spread(async (cat) => {
+      const { id: category } = cat.get();
```
Comment (MemunaHaruna) [inline src/api/controllers/articleController.js]: @murediane what does `cat` mean?

File: src/api/controllers/articleController.js [hunk 27]
```diff
@@ -0,0 +1,50 @@
...
+   * register a new
+   * @static
+   * @param {*} req the request
+   * @param {*} res the response to be sent
+   * @memberof ArticleController
+   * @returns {Object} res
+   */
+  static async create(req, res) {
+    const { id: author } = req.user;
+    const valid = await articleValidator(req.body);
```
Comment (MemunaHaruna) [inline src/api/controllers/articleController.js]: @murediane I think `isArticleValid` is a better variable name than `valid`

File: src/api/controllers/articleController.js [hunk 28]
```diff
@@ -0,0 +1,50 @@
+import cloudinary from 'cloudinary';
+import models from '../models/index';
+import statusCodes from '../../helpers/constants/status.codes';
+import errorMessage from '../../helpers/constants/error.messages';
+import articleValidator from '../../helpers/validators/articleValidator';
+
+const { Article, Category } = models;
+const { CREATED } = statusCodes;
```
Comment (MemunaHaruna) [inline src/api/controllers/articleController.js:9]: @murediane why use `CREATED` when you're getting a list of status codes?

File: src/api/controllers/articleController.js [hunk 29]
```diff
@@ -0,0 +1,50 @@
...
+    const { id: author } = req.user;
+    const valid = await articleValidator(req.body);
+    const { category: name } = req.body;
+    let image = '';
+
+    if (req.file) {
+      const saved = await cloudinary.v2.uploader.upload(req.file.path);
+      image = saved.secure_url;
+    }
+    await Category.findOrCreate({ where: { name } }).spread(async (cat) => {
```
Comment (MemunaHaruna) [inline src/api/controllers/articleController.js]: @murediane I don't think we want to allow users to create their own category. There should be a set list of categories that the app supports.

File: src/api/controllers/articleController.js [hunk 30]
```diff
@@ -0,0 +1,50 @@
...
+      const { id: category } = cat.get();
+      const article = await Article.create({
+        ...valid,
+        slug: '',
+        author,
+        category,
+        image
+      });
+      if (article) {
```
Comment (MemunaHaruna) [inline src/api/controllers/articleController.js]: @murediane what happens if this condition `if (article)` returns false?

File: src/api/migrations/20190622195902-create-article.js [hunk 31]
```diff
@@ -0,0 +1,34 @@
+export default {
+  up: (queryInterface, Sequelize) => queryInterface.createTable('articles', {
```
Comment (MemunaHaruna) [inline src/api/migrations/20190622195902-create-article.js]: @murediane `Article` not `articles`

File: src/api/models/article.js [hunk 32]
```diff
@@ -0,0 +1,57 @@
...
+      },
+      category: {
+        allowNull: false,
+        type: DataTypes.INTEGER,
+        references: { model: 'categories', key: 'id' }
+      }
+    },
+    {
+      tableName: 'articles',
+      paranoid: true,
```
Comment (MemunaHaruna) [inline src/api/models/article.js:40]: @murediane what does `paranoid: true` do?

File: src/api/models/user.js [hunk 33]
```diff
@@ -1,5 +1,5 @@
 export default (sequelize, DataTypes) => {
-  const User = sequelize.define('User',
+  const User = sequelize.define('Users',
```
Comment (MemunaHaruna) [inline src/api/models/user.js:2]: @murediane why do you have changes in this file?

File: src/api/seeders/categories.js [hunk 34]
```diff
@@ -0,0 +1,31 @@
+const moment = require('moment');
+
+const createdAt = moment().format();
+const updatedAt = createdAt;
+
+module.exports = {
```
Comment (MemunaHaruna) [inline src/api/seeders/categories.js]: @murediane These categories you've seeded are too few. Please ask the PO for a list of categories. You can also use the categories found on Medium.

File: src/configs/environments.js [hunk 35]
```diff
@@ -84,5 +84,4 @@ const environments = [
 ];
 
 const currentEnv = environments.find(el => el.name === env.toLocaleLowerCase());
```
Comment (MemunaHaruna) [inline src/configs/environments.js:86]: @murediane remove your changes in this file.

File: src/index.js [hunk 36]
```diff
@@ -1,18 +1,35 @@
...
+import session from 'express-session';
+import cors from 'cors';
+import logger from 'morgan';
 import apiRouter from './api/routes/index';
 import docsRouter from './api/routes/docs';
 import homeRouter from './api/routes/home';
-import register from './middlewares/register.app';
 import env from './configs/environments';
 
+import passportConfig from './middlewares/passport';
```
Comment (MemunaHaruna) [inline src/index.js]: @murediane why do you need this?

File: src/middlewares/ValidateUser.js [hunk 37]
```diff
@@ -8,11 +8,11 @@ const { User } = models;
  */
 export default class {
   /**
-  * @param  {object} req The request object containing method, body, params, query,...ß
-  * @param  {object} res The response object for the request
-  * @param  {function} next The next middleware function in the request pipeline
-  * @returns {object} The response with status and data or error message
-  */
+   * @param  {object} req The request object containing method, body, params, query,...ß
```
Comment (MemunaHaruna) [inline src/middlewares/ValidateUser.js:11]: @murediane undo your changes in this file.

**Review by MemunaHaruna — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 25 above)
Comment (murediane) [reply src/index.js]: We need it for set response cookies and read from the response.
you can read more about on https://www.npmjs.com/package/express-session

File: src/api/models/index.js (hunk 23 above)
Comment (murediane) [reply src/api/models/index.js]: so I should delete it right?I am not using it I just rebased.

File: src/api/models/article.js (hunk 20 above)
Comment (murediane) [reply src/api/models/article.js:6]: Can you explain what this mean

File: src/api/migrations/20190622195901-create-category.js (hunk 17 above)
Comment (murediane) [reply src/api/migrations/20190622195901-create-category.js:3]: done

File: src/index.js [hunk 38]
```diff
@@ -1,18 +1,38 @@
...
+app.use(logger('dev'));
+
+app.use(express.static(`${__dirname}/public`));
+app.use(passport.initialize());
+passportConfig(passport);
+
+app.use(session({
+  secret: env.secret,
+  cookie: { maxAge: 60000 },
+  resave: false,
```
Comment (MemunaHaruna) [inline src/index.js]: @murediane why are you setting these cookies?

Comment (murediane) [reply src/index.js]: They are removed now.

File: src/api/controllers/articleController.js (hunk 1 above)
Comment (murediane) [reply src/api/controllers/articleController.js:25]: yes

File: src/api/controllers/articleController.js (hunk 4 above)
Comment (premlogic) [reply src/api/controllers/articleController.js]: yeah. It's better

File: package.json [hunk 39]
```diff
@@ -5,12 +5,14 @@
...
+    "test": "npm run setup:test && NODE_ENV=test nyc --reporter=html --reporter=text --reporter=lcov ./node_modules/.bin/mocha  ./tests/* --require @babel/register --timeout 5000 --exit",
     "dev": "NODE_ENV=development nodemon --exec babel-node src/index.js",
-    "coverage": "nyc report --reporter=text-lcov | coveralls",
-    "eslint": "eslint . --cache --fix",
     "setup:test": "NODE_ENV=test npm run migrate:undo  && NODE_ENV=test npm run migrate && NODE_ENV=test npm run seed",
     "setup:dev": "NODE_ENV=development npm run migrate:undo && NODE_ENV=development npm run migrate && NODE_ENV=development npm run seed",
+    "pretest": "NODE_ENV=test sequelize db:migrate:undo:all && NODE_ENV=test sequelize db:migrate",
+    "coverage": "nyc report --reporter=text-lcov | coveralls",
+    "eslint": "eslint . --cache --fix",
+    "setup": "NODE_ENV=test npm run migrate:undo  && NODE_ENV=test npm run migrate && NODE_ENV=test npm run seed",
```
Comment (MemunaHaruna) [inline package.json:15]: @murediane what's the difference between the `setup` which you've added here and the already existing `setup:test`?

File: src/api/controllers/articleController.js (hunk 1 above)
Comment (MemunaHaruna) [reply src/api/controllers/articleController.js:25]: @murediane I think @nkpremices was correcting the comment `register a new` here.

File: src/api/controllers/articleController.js [hunk 40]
```diff
@@ -1,17 +1,61 @@
...
+   * @static
+   * @param {*} req the request
+   * @param {*} res the response to be sent
+   * @memberof ArticleController
+   * @returns {Object} res
+   */
+  static async create(req, res) {
+    const { id: author } = req.user.user;
+
+    const isArticleValid = await articleValidator(req.body);
```
Comment (MemunaHaruna) [inline src/api/controllers/articleController.js]: @murediane does this `isArticleValid` hold a boolean value? The name suggests so.

File: src/api/controllers/articleController.js [hunk 41]
```diff
@@ -25,7 +69,7 @@ class ArticleController {
   static async updateArticle(req, res) {
     const { slug } = req.params;
     const {
-      title, description, body, tagList, category
+      title, description, body, category
```
Comment (MemunaHaruna) [inline src/api/controllers/articleController.js]: @murediane did @MCFrank16 okay the removal of this `tagList`?

File: src/api/controllers/authController.js (hunk 5 above)
Comment (MemunaHaruna) [reply src/api/controllers/authController.js:39]: @murediane please, remove this file.

File: src/api/controllers/verifyEmail.js [hunk 42]
```diff
@@ -8,20 +8,21 @@ import errorMessages from '../../helpers/constants/error.messages';
  */
 export default class {
   /**
```
Comment (MemunaHaruna) [inline src/api/controllers/verifyEmail.js:10]: @murediane please, undo your changes in this file. It's not related to your work.

File: src/api/migrations/20190622195901-create-category.js (hunk 17 above)
Comment (MemunaHaruna) [reply src/api/migrations/20190622195901-create-category.js:3]: @murediane I see that you're renaming the create-token-table migration. Please, don't edit migration files like this.  Leave the token migration file exactly the way it is (check Develop branch) and create a separate migration file for categories.

File: src/api/migrations/20190625141140-drop-token-table.js [hunk 43]
```diff
@@ -1,23 +0,0 @@
-export default {
-  up: (queryInterface, Sequelize) => queryInterface.dropTable('Tokens'),
```
Comment (MemunaHaruna) [inline src/api/migrations/20190625141140-drop-token-table.js]: @murediane why have you deleted this file? Please, no deleting or editing of migration files.

File: src/api/models/article.js (hunk 20 above)
Comment (MemunaHaruna) [reply src/api/models/article.js:6]: @murediane variable names should be meaningful and descriptive so that others reading our code understand the context behind the variable.

File: src/api/models/article.js [hunk 44]
```diff
@@ -1,59 +1,58 @@
...
-        allowNull: false
-      },
-      userId: {
         type: DataTypes.INTEGER,
-        references: {
-          model: 'Users',
-          key: 'id'
-        },
-        onDelete: 'CASCADE',
-        onUpdate: 'CASCADE',
```
Comment (MemunaHaruna) [inline src/api/models/article.js:47]: @murediane why have you removed these:

onDelete: 'CASCADE',	
onUpdate: 'CASCADE',

File: src/api/routes/articles.js [hunk 45]
```diff
@@ -0,0 +1,14 @@
+// import { Router } from 'express';
+// import checkAuth from '../../middlewares/checkAuth';
```
Comment (MemunaHaruna) [inline src/api/routes/articles.js]: @murediane remove commented code.

File: src/api/routes/signup.js [hunk 46]
```diff
@@ -1,8 +0,0 @@
-import router from 'express';
```
Comment (MemunaHaruna) [inline src/api/routes/signup.js:1]: @murediane why did you delete this file?

File: tests/updateArticle.test.js [hunk 47]
```diff
@@ -96,33 +96,33 @@ describe('testing testing testing and testing for the article update controller'
...
-        expect(res.body)
-          .to.have.property('message')
-          .eql('Your Article is up-to-date now, Thanks');
-        done();
-      });
-  });
+  // it('should update the article with new content', (done) => {
+  //   const slug = 'How-to-create-sequalize-seeds';
+  //   chai
+  //     .request(app)
```
Comment (MemunaHaruna) [inline tests/updateArticle.test.js]: @murediane remove unused commented code

File: tests/article.js [hunk 48]
```diff
@@ -0,0 +1,112 @@
...
+      .set('authorization', userToken)
+      .send(rest)
+      .end((err, res) => {
+        const { status, body } = res;
+        expect(status).to.equal(statusCode.BAD_REQUEST);
+        expect(body).to.have.property('errors');
+        done();
+      });
+  });
+  it('should not post an article without body', (done) => {
```
Comment (MemunaHaruna) [inline tests/article.js:98]: @murediane is there a test for when an article is successfully created?

**Review by MemunaHaruna — APPROVED** [review]

File: package.json (hunk 14 above)
Comment (murediane) [reply package.json:9]: This is the one for develop I updated .The changes are showing the previous script from my branch. Check develop you will realize it is the same script

File: src/api/migrations/20190622195901-create-category.js (hunk 17 above)
Comment (murediane) [reply src/api/migrations/20190622195901-create-category.js:3]: I was requested removed it from the previous request and it seems like there was a misunderstanding.  We discussed about it today  they should be there .Let me bring them back

File: src/api/models/article.js (hunk 20 above)
Comment (murediane) [reply src/api/models/article.js:6]: There is no meaning for the variable apart from it being a number . I was avoiding a no magical  number error .because I am just using those number to get the help to generate the slug but they have no specific role that can be understood.they are just numbers

File: src/api/controllers/articleController.js (hunk 40 above)
Comment (murediane) [reply src/api/controllers/articleController.js]: it does not contain boolean .it  contains an object. I renamed as you suggested in previous comment

File: src/api/controllers/articleController.js (hunk 41 above)
Comment (murediane) [reply src/api/controllers/articleController.js]: yes he was

File: src/api/models/article.js (hunk 44 above)
Comment (murediane) [reply src/api/models/article.js:47]: They are not removed  Frank did not have a relationship with categories model so I had to move them to the appropriate place. but they are still in the file

File: src/api/routes/signup.js (hunk 46 above)
Comment (murediane) [reply src/api/routes/signup.js:1]: It was the route to signup that was just used to validate data and it is no longer needed since the middleware is called in the signup route. It was just useless

File: tests/updateArticle.test.js (hunk 47 above)
Comment (murediane) [reply tests/updateArticle.test.js]: alright, sorry let me remove it

File: package.json (hunk 39 above)
Comment (MemunaHaruna) [reply package.json:15]: @murediane I can still see `setup` and `setup:test` here
